osquery/tables/system/windows/smbios_tables.cpp

```diff
@@ -33,6 +34,49 @@ std::string to_iso8601_date(const FILETIME& ft) {
   return iso_date.str();
 }
 
+std::string getFirmwareType() {
+  enum class FirmwareType : std::uint32_t {
+    Unknown,
+    Bios,
+    Uefi,
+  };
+  using GetFirmwareTypePtr = BOOL (*)(FirmwareType*);
+
+  auto kernel32_module = GetModuleHandleA("kernel32");
+  auto function_ptr = reinterpret_cast<GetFirmwareTypePtr>(
+      GetProcAddress(kernel32_module, "GetFirmwareType"));
+  FirmwareType firmware_type = FirmwareType::Unknown;
+
+  if (function_ptr == nullptr) {
+    // We are on Windows 7: Attempt to determine the firmware type based on
+    // the registry keys
+    HKEY state_reg_key;
+    auto success =
+        RegCreateKeyA(HKEY_LOCAL_MACHINE,
+                      "SYSTEM\\CurrentControlSet\\Control\\SecureBoot\\State",
+                      &state_reg_key);
+
+    if (success != 0 && state_reg_key != INVALID_HANDLE_VALUE) {
+      firmware_type = FirmwareType::Uefi;
+      RegCloseKey(state_reg_key);
+
+    } else {
+      firmware_type = FirmwareType::Bios;
+    }
+
+  } else if (!function_ptr(&firmware_type)) {
+    LOG(ERROR) << "platform_info: Failed to acquire the firmware type";
+  }
+
+  if (firmware_type == FirmwareType::Bios) {
+    return "Bios";
+  } else if (firmware_type == FirmwareType::Uefi) {
+    return "Uefi";
+  }
+
+  return "Unknown";
```

Comment:
Title case feels weird to me. I'd probably prefer lower.

Reply:
Done.